core/commands/config.go

```diff
@@ -278,6 +279,88 @@ can't be undone.
 	},
 }
 
+var configProfileCmd = &cmds.Command{
+	Helptext: cmds.HelpText{
+		Tagline: "Apply profiles to config.",
+	},
+
+	Subcommands: map[string]*cmds.Command{
+		"apply":  configProfileApplyCmd,
+		"revert": configProfileRevertCmd,
+	},
+}
+
+var configProfileApplyCmd = &cmds.Command{
+	Helptext: cmds.HelpText{
+		Tagline: "Apply profile to config.",
+	},
+	Arguments: []cmds.Argument{
+		cmds.StringArg("profile", true, false, "The profile to apply to the config."),
+	},
+	Run: func(req cmds.Request, res cmds.Response) {
+		args := req.Arguments()
+
+		profile, ok := config.Profiles[args[0]]
+		if !ok {
+			res.SetError(fmt.Errorf("%s in not a profile", args[0]), cmds.ErrNormal)
+			return
+		}
+
+		err := transformConfig(req.InvocContext().ConfigRoot, profile.Apply)
+		if err != nil {
+			res.SetError(err, cmds.ErrNormal)
+			return
+		}
+	},
+}
+
+var configProfileRevertCmd = &cmds.Command{
+	Helptext: cmds.HelpText{
+		Tagline: "Revert profile changes.",
+		ShortDescription: `Reverts profile-related changes to the config.
+
+Reverting some profiles may damage the configuration or not be possible.
+Backing up the config before running this command is advised.`,
+	},
+	Arguments: []cmds.Argument{
+		cmds.StringArg("profile", true, false, "The profile to apply to the config."),
+	},
+	Run: func(req cmds.Request, res cmds.Response) {
+
+		profile, ok := config.Profiles[req.Arguments()[0]]
+		if !ok {
+			res.SetError(fmt.Errorf("%s in not a profile", req.Arguments()[0]), cmds.ErrNormal)
+			return
+		}
+
+		err := transformConfig(req.InvocContext().ConfigRoot, profile.Unapply)
+		if err != nil {
+			res.SetError(err, cmds.ErrNormal)
+			return
+		}
+	},
+}
+
+func transformConfig(configRoot string, transformer config.Transformer) error {
+	r, err := fsrepo.Open(configRoot)
```

Comment:
if the daemon is running while the user runs this they might run into issues. You cant open the repo again

Reply:
probably best to pass in the repo from the IpfsNode

Reply:
That's what other commands here do. Looking at repo.Open it should be safe:
```go
func Open(repoPath string) (repo.Repo, error) {
	fn := func() (repo.Repo, error) {
		return open(repoPath)
	}
	return onlyOne.Open(repoPath, fn)
}
```

Reply:
oh, i see. good point.